pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -308,28 +382,37 @@ func IsNodeWithLowUtilization(nodeThresholds api.ResourceThresholds, thresholds
 	return true
 }
 
-func NodeUtilization(node *v1.Node, pods []*v1.Pod) (api.ResourceThresholds, []*v1.Pod, []*v1.Pod, []*v1.Pod, []*v1.Pod, []*v1.Pod) {
```

Comment:
I am still trying to understand why you have split this function "NodeUtilization" into 2 functions (classifyPods and NodeUtilization). The reason I had combined them for "performance reasons". I wanted to traverse all nodes and their pods "just once". So please make sure you are not traversing the lists of pods more than once. Try to create anything just based on one traversal of all pods on all nodes.

Reply:
I thought about that when I was making changes. If you think performance is a bottleneck, I can revert the change but I was trying to make code a bit more readable.

Reply:
> If you think performance is a bottleneck, I can revert the change but I was trying to make code a bit more readable.

This code is not really large that we can not maintain readability. You could always split that function into multiple functions but can call new functions inside the bigger function to improve readability, You do not have to separate them.

Reply:
Its more on the lines of one function doing one thing but I see your point. I can make the changes after writing the benchmark.